idseq_dag/engine/pipeline_flow.py

```diff
@@ -79,6 +79,22 @@ def parse_and_validate_conf(dag_json):
           "given_targets": input files that are given
         '''
         dag = json.loads(open(dag_json).read())
+
+        # hack -- add some targets here to support cd-hit-dup work
+        cdhitdup_cluster_sizes_path = "cdhitdup_cluster_sizes.tsv"
+        cdhitdup_cluster_sizes_target = "cdhitdup_cluster_sizes"
+        if dag["name"] in ("postprocess", "non_host_alignment"):
+            dag["targets"][cdhitdup_cluster_sizes_target] = [
+                cdhitdup_cluster_sizes_path
+            ]
+            dag["given_targets"][cdhitdup_cluster_sizes_target] = {
+                "s3_dir": dag["given_targets"]["host_filter_out"]["s3_dir"]
+            }
+        if dag["name"] == "host_filter":
+            for fff in ["dedup1.fa.clstr", cdhitdup_cluster_sizes_path]:
+                if fff not in dag["targets"]["cdhitdup_out"]:
+                    dag["targets"]["cdhitdup_out"].append(fff)
+
```

Comment:
for clarity, I'd like to see these lines extracted to a private method, and linked to a JIRA ticket that tracks the clean up work needed

Reply:
I believe these targets need to be added to the DAG json in idseq-web, otherwise it's incompatible with the new SFN-WDL infra -- @kislyuk is that correct? Seems like the current code would require some ad-hoc migration

Reply:
Yes, first we merge this PR into idseq dag, then we deploy an idseq-web change to emit correct DAG, then we remove this hack from idseq-dag.

Reply:
I've updated the main PR description with the resolutions to all resolved questions, current list of todos, and most importantly, the deployment plan.

Reply:
The hack lines above are really not meant to live in master for more than 1-2 weeks tops.  Basically they can be reverted here as soon as we deploy the idseq-web change that does the same (with may be a couple extra days to make sure we won't have to revert that change).